### ISSUE TITLE
using `in` with an empty array pattern produces invalid SQL

### ISSUE THREAD

**Issue by Globidev** [PRQL/prql#4595]
### What happened?

`in [] value` transpiles to `value in ()` which is not valid syntax (at least in postgresql)

### PRQL input

```elm
prql target:sql.postgres
from table
filter (field | in [])
```

### SQL output

```SQL
SELECT
  *
FROM
  "table"
WHERE
  field IN ()
```

### Expected SQL output

```SQL
SELECT
  *
FROM
  "table"
WHERE
  FALSE -- Or maybe some other smart alternative I don't know about
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

### Anything else?

Thanks for this wonderful project ❤️

**Comment by max-sixty** [comment]
Thanks @Globidev . Indeed, seems like we should we working around that postgres issue. (Ideally postgres would allow an empty list!).

This would likely be a fairly small PR into https://github.com/PRQL/prql/blob/2f2c08b168e9f6894ff65e315da44c6a86e6237d/prqlc/prqlc/src/sql/gen_expr.rs#L159-L186 if anyone wants to give it a go.

**Comment by PrettyWood** [comment]
Closed by #4598
